constraintlayout/core/src/main/java/androidx/constraintlayout/core/state/TransitionParser.java

```diff
@@ -321,6 +330,71 @@ private static void parseKeyAttribute(CLObject keyAttribute,
                 }
             }
         }
+        // Support for custom attributes in KeyAttributes
+        CLElement customElement = keyAttribute.getOrNull("custom");
+        if (customElement != null && customElement instanceof CLObject) {
+            CLObject customObj = ((CLObject) customElement);
```

Comment:
Not sure if the outside parenthesis is necessary here.
(CLObject) customElement should also work?